smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/HttpBindingProtocolGenerator.kt

```diff
@@ -403,6 +405,31 @@ abstract class HttpBindingProtocolGenerator : ProtocolGenerator {
         return resolved
     }
 
+    // Checks for @requiresLength trait
+    // Returns true if the operation:
+    // - has a streaming member with @httpPayload trait
+    // - target is a blob shape with @requiresLength trait
+    private fun hasRequiresLengthTrait(ctx: ProtocolGenerator.GenerationContext, op: OperationShape): Boolean {
+        if (op.input.isPresent) {
+            val inputShape = ctx.model.expectShape(op.input.get())
+            val streamingMember = inputShape.findStreamingMember(ctx.model)
+            if (streamingMember != null) {
+                val targetShape = ctx.model.expectShape(streamingMember.target)
+                if (targetShape != null) {
+                    return streamingMember.hasTrait<HttpPayloadTrait>() &&
+                        targetShape.isBlobShape &&
+                        targetShape.hasTrait<RequiresLengthTrait>()
+                }
+            }
+        }
+        return false
+    }
+
+    // Checks for @unsignedPayload trait on an operation
+    private fun hasUnsignedBody(op: OperationShape): Boolean {
```

Comment:
nit: Maybe rename the function to `hasUnsignedBodyTrait` for uniformity with `hasRequiresLengthTrait` and more context with just one more word

Reply:
Or is this even needed? It does make it read a bit more smoother, but I'm unsure whether introducing a new method here is a good idea given the readability and amount of functionality provided by the function doesn't seem to outweigh the extra bandwidth in having to remember to use this function instead of just calling `hasTrait<[TraitName]>()`

Reply:
Changing to `hasUnsignedPayloadTrait`. I think I like it in a helper for readability + maintainability should the condition change. Keeping logic inside of where we append/instantiate middlewares seemed a bit hairy